cpp/include/cudf/quantiles.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.hpp"
+
+namespace cudf {
+
+std::pair<cudf::table, cudf::table>
+group_quantiles(cudf::table const& key_table,
+                cudf::table const& val_table,
+                std::vector<double> const& quantiles,
+                gdf_quantile_method interpolation = GDF_QUANT_LINEAR,
+                gdf_context const& context = gdf_context{});
```

Comment:
Echo Jake's concern about gdf_context. I never liked the name, I want it to change eventually. Let's avoid it.